kubespawner/spawner.py

```diff
@@ -1842,7 +1842,8 @@ def _start(self):
                     self.pod_name,
                 )
                 self._start_watching_pods(replace=True)
-            raise
+            else:
+                raise
```

Comment:
It could be that we reraise the exception because even after restarting the reflector this particular attempt at spawning the pod won't recover. That is speculation though.

Maybe @yuvipanda or @manics know why the code is the way it is?

Reply:
Sorry, I've no idea. The original code is from @minrk https://github.com/jupyterhub/kubespawner/commit/42a48acdbe4471562549307d0d0dff91e8a1b582

Reply:
I think the original code make sense, but the change doesn't.

My understanding is that _start is supposed to fail with an error if it fails, hence `raise`. The logic about restarting the pod reflector here is to ensure that the Singleton reflector (keeps track of pods in the namespace for all spawner instances) isn't stuck in a disfunctional state by restarting it.